pilot/pkg/model/xds_cache.go

```diff
@@ -15,13 +15,47 @@
 package model
 
 import (
+	"fmt"
+	"istio.io/pkg/monitoring"
 	"sync"
 
 	"github.com/golang/protobuf/ptypes/any"
 
+	"github.com/hashicorp/golang-lru"
+
 	"istio.io/istio/pilot/pkg/util/sets"
 )
 
+var (
+	writes = monitoring.NewSum(
+		"xds_cache_writes",
+		"Total number of xds cache writes.",
+		monitoring.WithLabels(typeTag),
+	)
+
+	reads = monitoring.NewSum(
+		"xds_cache_reads",
+		"Total number of xds cache reads.",
+		monitoring.WithLabels(typeTag),
+	)
+
+	hits   = reads.With(typeTag.Value("hit"))
+	misses = reads.With(typeTag.Value("hit"))
+)
+
+// TODO tune this and make configurable
```

Comment:
This is way to small. We will blow through this immediately only any small size cluster

Reply:
Also lets make it configurable prior to merge. Users on big clusters probably want this pretty high. Ideally it would be dynamic as well but not sure how we would do that

Reply:
The lib supports resizing. We could implement some logic around seeing "n evictions in window sized t", bump the cache size.

Reply:
```go
// in evict callback
atomic.AddInt32((*int32)(c), 1)
if atomic.LoadInt32((*int32)(c)) > 100 {
  l.lru.Resize(current + 100)
}
time.AfterFunc(time.Minute, func() {
  atomic.AddInt32((*int32)(c), -1)
})
```